europi.go

```diff
@@ -5,60 +5,107 @@ import (
 )
 
 const (
+	// EuroPi voltage range constants.
 	MaxVoltage = 10.0
 	MinVoltage = 0.0
+
+	// EuroPi hardware GPIO pins.
+	DIPin = machine.GPIO22
+	AIPin = machine.ADC0
+
+	DisplaySdaPin = machine.GPIO0
+	DisplaySclPin = machine.GPIO1
+
+	K1Pin = machine.ADC1
+	K2Pin = machine.ADC2
+
+	B1Pin = machine.GPIO4
+	B2Pin = machine.GPIO5
+
+	CV1Pin = machine.GPIO21
+	CV2Pin = machine.GPIO20
+	CV3Pin = machine.GPIO16
+	CV4Pin = machine.GPIO17
+	CV5Pin = machine.GPIO18
+	CV6Pin = machine.GPIO19
+)
+
+var (
+	DisplayChannel = machine.I2C0
+	CV1PwmGroup    = machine.PWM2
+	CV2PwmGroup    = machine.PWM2
+	CV3PwmGroup    = machine.PWM0
+	CV4PwmGroup    = machine.PWM0
+	CV5PwmGroup    = machine.PWM1
+	CV6PwmGroup    = machine.PWM1
+
+	europi *EuroPi
 )
 
 // EuroPi is the collection of component wrappers used to interact with the module.
 type EuroPi struct {
-	// Display is a wrapper around ssd1306.Device
-	Display *Display
+	// Display provides methods for drawing to the OLED display.
+	Display *display
+
+	DI *digitalInput
+	// AI provides methods for reading analog input control voltage between 0 and 12V.
+	AI *analogInput
 
-	DI DigitalReader
-	AI AnalogReader
+	// B1 is a struct for handling the left push button behavior.
+	B1 *button
+	// B2 is a struct for handling the right push button behavior.
+	B2 *button
 
-	B1 DigitalReader
-	B2 DigitalReader
+	// K1 provides methods for reading knob voltage and position for the left knob.
+	K1 *knob
+	// K2 provides methods for reading knob voltage and position for the left knob.
+	K2 *knob
 
-	K1 AnalogReader
-	K2 AnalogReader
+	// CV1-6 are structs for interacting with the cv output jacks.
+	CV1 *output
+	CV2 *output
+	CV3 *output
+	CV4 *output
+	CV5 *output
+	CV6 *output
+	// CV is an array containing all CV outputs.
+	CV [6]*output
+}
 
-	CV1 Outputer
-	CV2 Outputer
-	CV3 Outputer
-	CV4 Outputer
-	CV5 Outputer
-	CV6 Outputer
-	CV  [6]Outputer
+func init() {
+	europi = new()
 }
 
-// New will return a new EuroPi struct.
-func New() *EuroPi {
```

Comment:
If you choose to return to New (see comment on singletons below), then we could get some neat function-callback-configuration parameter thing going on. [This](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) is what I'm talking about. So something like

```go
knobNumber := 0
ep := europi.New(
    europi.WithKnob(knobNumber, mockADC), // each of these package level option functions returns *another* function!
    europi.WithButton(buttonNumber, mockDigitalReader),
    europi.WithDisplay(myHDDisplay),
)
```
This would be pretty neat since your API would then stay pretty much the same into the future, and if a user wanted to initialize it ez-pz they could just go `ep := europi.New()`